crates/oxc_linter/src/rules/unicorn/no_array_method_this_argument.rs

```diff
@@ -0,0 +1,317 @@
+        .first()
+        .is_some_and(|arg| arg.as_expression().is_none_or(|expr| is_node_not_function(expr)))
+        || call_expr.arguments.get(1).is_some_and(|arg| matches!(arg, Argument::SpreadElement(_)))
+        || is_ignored_expr(&call_expr.callee)
+    {
+        return;
+    }
+
+    ctx.diagnostic(no_array_method_this_argument_diagnostic(
+        call_expr.arguments.first().map_or(call_expr.span, GetSpan::span),
+    ));
+}
+
+fn check_array_from<'a>(node: &AstNode<'a>, ctx: &LintContext<'a>) {
+    let AstKind::CallExpression(call_expr) = node.kind() else { return };
+
+    if !is_method_call(call_expr, Some(&["Array"]), Some(&["from", "fromAsync"]), Some(3), Some(3))
+        || call_expr.arguments.first().is_some_and(|arg| matches!(arg, Argument::SpreadElement(_)))
+        || call_expr.arguments.get(2).is_some_and(|arg| matches!(arg, Argument::SpreadElement(_)))
+        || call_expr
+            .arguments
+            .get(1)
+            .is_some_and(|arg| arg.as_expression().is_none_or(|expr| is_node_not_function(expr)))
+    {
+        return;
+    }
+
+    ctx.diagnostic(no_array_method_this_argument_diagnostic(
+        call_expr.arguments.get(2).map_or(call_expr.span, GetSpan::span),
+    ));
+}
+
+fn is_ignored_expr(mut expr: &Expression) -> bool {
+    let mut path = Vec::new();
+
+    while let Some(member_expr) = expr.as_member_expression() {
+        let MemberExpression::StaticMemberExpression(static_mem_expr) = member_expr else {
+            return false;
+        };
+        path.push(static_mem_expr.property.name.as_str());
+
+        expr = member_expr.object().get_inner_expression();
+    }
+
+    let Expression::Identifier(ident) = expr else { return false };
+    path.push(ident.name.as_str());
+    let path = path.iter().rev();
+
+    for e in IGNORED {
+        if e.iter().zip(path.clone()).all(|(x, y)| x == y) {
+            return true;
+        }
+    }
+
+    false
+}
+
+fn is_node_not_function(expr: &Expression) -> bool {
+    match expr {
+        Expression::ArrayExpression(_)
+        | Expression::BinaryExpression(_)
+        | Expression::ClassExpression(_)
+        | Expression::NullLiteral(_)
+        | Expression::RegExpLiteral(_)
+        | Expression::StringLiteral(_)
+        | Expression::BigIntLiteral(_)
+        | Expression::NumericLiteral(_)
+        | Expression::TemplateLiteral(_)
+        | Expression::UnaryExpression(_)
+        | Expression::UpdateExpression(_)
+        | Expression::AssignmentExpression(_)
+        | Expression::AwaitExpression(_)
+        | Expression::NewExpression(_)
+        | Expression::TaggedTemplateExpression(_)
+        | Expression::ThisExpression(_) => true,
+        Expression::Identifier(ident) if ident.name == "undefined" => true,
+        Expression::CallExpression(call_expr) => {
+            !is_method_call(call_expr, None, Some(&["bind"]), None, None)
+        }
+        _ => false,
+    }
+}
+
+const IGNORED: &[&[&str]] = &[
```

Comment:
Is it possible to check in `check_array_prototype_methods` for the `Array` object and avoiding this ignore list?
Like we are doing it for `check_array_from`

Reply:
Nevermind. These are `__proto__` functions